### PR TITLE
re #17486 - set _locationManager if auth skipped

### LINKED ISSUE #17486: Geolocation times out when using "skipPermissionRequests"
### Is this a bug report?

Yes 

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

 Environment:
  OS: macOS Sierra 10.12.6
  Node: 7.0.0
  Yarn: Not Found
  npm: 3.10.8
  Watchman: 4.7.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 2.3 AI-162.3871768

Packages: (wanted => installed)
  react: ^16.0.0 => 16.2.0
  react-native: 0.51.0 => 0.51.0

### Steps to Reproduce

1. Add navigator.geolocation.setRNConfiguration({'skipPermissionRequests':true}); to your code. I added it to the constructor or componentWillMount for the app.
2. Run application and try geolocating 
3. Geolocation will timeout.

### Expected Behavior

Geolocation will work.

### Actual Behavior

Geolocation times out.

### Reproducible Demo

Take any existing / working geolocation demo and add the line in step #1. 

### Potential Solution
As I stated in #15096 I believe this is an issue with the fact that _locationManager is being set inside the Authorization method which gets skipped. Because of this the _locationManager is not set and so just times out.

Adding the following to beginLocationUpdatesWithDesiredAccuracy fixes the issue as now locationManager is set again.

if (!_locationManager) {
    _locationManager = [CLLocationManager new];
    _locationManager.delegate = self;
}

### Contributors that are probably familiar with that code
Contributor who added the feature via #15096 - @rigdern (Thanks!!) 
Code Reviewer - @javache 
Code Reviewer / Merger - @hramos

### PR DESCRIPTION
**Pull request by ngandhy**
This fixes #17486

make sure locationManager is being set before continuing. 

Used if !locationManager vs else on previous statement as we should NEVER enter this code without _locationManager set. Also the else version might experience issues if someone touches the auth code and doesn't check this case, so seems more "long term stable".

## Motivation

This fixes #17486

## Test Plan

1) Have a working geolocation demo
2) Add navigator.geolocation.setRNConfiguration({'skipPermissionRequests':true}); to your code. I added it to the constructor or componentWillMount for the app.
3) Observe that geolocation no longer works (times out)
4) Apply patch
5) Observe that geolocation works again

## Related PRs

re #15096

## Release Notes

[IOS] [BUGFIX] [GeoLocation] - Fix skipPermissionRequests by setting _locationManager

### CODE REVIEW

**Comment by ngandhy** [comment]
@rigdern - please do lmk if you have any feedback on the change. I thought this was the lowest touch way to address it.

File: Libraries/Geolocation/RCTLocationObserver.m [hunk 1]
```diff
@@ -158,6 +158,12 @@ - (void)beginLocationUpdatesWithDesiredAccuracy:(CLLocationAccuracy)desiredAccur
   if (!_locationConfiguration.skipPermissionRequests) {
     [self requestAuthorization];
   }
+  
+  //re #17486 - If someone skips permissions, location manager is not being set. so lets make sure its set before continuing
```
Comment (rigdern) [inline Libraries/Geolocation/RCTLocationObserver.m]: I don't think this comment is necessary. The code is clear enough without it.

**Comment by ngandhy** [comment]
@rigdern - my bad! I could swear I tagged you but must have done something wrong. I'll make sure to do that next time.

Great! I made the change you suggested as well :)

**Comment by ide** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**